presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -554,17 +554,38 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         }
 
         // 4. Project and re-write all grouping functions
+        return handleGroupingOperations(subPlan, node, groupIdSymbol);
+    }
+
+    private PlanBuilder handleGroupingOperations(PlanBuilder subPlan, QuerySpecification node, Optional<Symbol> groupIdSymbol)
+    {
         if (analysis.getGroupingOperations(node).isEmpty()) {
             return subPlan;
         }
-        return subPlan.appendProjections(
-                analysis.getGroupingOperations(node).stream().map(Expression.class::cast).collect(toImmutableList()),
-                symbolAllocator,
-                idAllocator,
-                Optional.of(node),
-                Optional.of(analysis),
-                Optional.of(metadata),
-                groupIdSymbol);
+
+        TranslationMap translations = subPlan.copyTranslations();
+
+        Assignments.Builder projections = Assignments.builder();
+
+        // Add an identity projection for the underlying plan
+        for (Symbol symbol : subPlan.getRoot().getOutputSymbols()) {
+            projections.put(symbol, symbol.toSymbolReference());
+        }
+
+        ImmutableMap.Builder<Symbol, Expression> newTranslations = ImmutableMap.builder();
```

Comment:
do you need this builder? I think you can directly add mappings to `translations`.